pandas/core/indexes/interval.py

```diff
@@ -1092,7 +1092,15 @@ def overlaps(self, other):
 
     def _setop(op_name):
         def func(self, other, sort=True):
-            other = self._as_like_interval_index(other)
+            try:
+                other = self._as_like_interval_index(other)
+            # dont catch ValueError so that mixed closed interval indexes raise
+            # only catch Non-Interval index mismatches.
+            except TypeError:
+                result = getattr(self.astype(object), op_name)(other)
```

Comment:
what is this all about?

Reply:
Since interval indexes don't inherit their set operation behavior from the parent Index class this enforces that incompatible types are treated as object dtype unions. This was reviewed by @jschendel who also made a couple suggestions about it.

Reply:
ok, its the op_name stuff that is a bit fishy to me, cc @jschendel

Reply:
Agreed that it's a bit fishy and kind of gross.  In part this is because we just have a single function that's trying to patch all of the set ops, so I don't find it particularly surprising that we're finding the need to special case things as more corner cases are discovered, especially since we're relying on non-interval specific implementations of the set ops.

Really, I'd like for these to have interval specific implementations, as I think that'd clean things up a bit, though probably out of scope here.  Right now things are done in a bit a round about way, as the set ops are done via IntervalIndex-->tuples-->MultiIndex-->SetOp or IntervalIndex-->Index[object]-->SetOp, which is certainly not optimal in terms of performance, and can be a bit lossy in terms of metadata.

Given that these could use custom implementations, I'm not too concerned about the code getting a little gross.  Certainly would want to curtail it as much as possible though.  Would more so want to make sure the tests are clear and extensive so that the necessary guardrails are in place when writing the custom implementations.